### PR TITLE
[systemtest] Update CO version in upgrade/downgrade files and environment class to 0.36.0

### PR DESCRIPTION
**Pull request by im-konge**
### Type of change

- Task

### Description

This PR updates CO version in STs to 0.36.0 after the release.

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

**Review by jankalinic — APPROVED** [review]
lgtm

**Review by see-quick — APPROVED** [review]

File: systemtest/src/test/resources/upgrade/BundleDowngrade.yaml [hunk 1]
```diff
@@ -26,18 +26,18 @@
...
   imagesAfterOperations:
-    zookeeper: strimzi/kafka:0.35.1-kafka-3.4.0
-    kafka: strimzi/kafka:0.35.1-kafka-3.4.0
-    topicOperator: strimzi/operator:0.35.1
-    userOperator: strimzi/operator:0.35.1
+    zookeeper: strimzi/kafka:0.36.0-kafka-3.4.0
+    kafka: strimzi/kafka:0.36.0-kafka-3.4.0
+    topicOperator: strimzi/operator:0.36.0
+    userOperator: strimzi/operator:0.36.0
   deployKafkaVersion: 3.4.0
```
Comment (scholzj) [inline systemtest/src/test/resources/upgrade/BundleDowngrade.yaml:41]: Should this be Kafka 3.5.0?

File: systemtest/src/test/resources/upgrade/BundleDowngrade.yaml [hunk 2]
```diff
@@ -49,18 +49,18 @@
...
```
Comment (scholzj) [inline systemtest/src/test/resources/upgrade/BundleDowngrade.yaml:64]: Same as above.

File: systemtest/src/test/resources/upgrade/BundleDowngrade.yaml (hunk 1 above)
Comment (im-konge) [reply systemtest/src/test/resources/upgrade/BundleDowngrade.yaml:41]: Yeah good catch, thanks.

Comment (scholzj) [reply systemtest/src/test/resources/upgrade/BundleDowngrade.yaml]: But should it be in the images as well?

**Review by scholzj — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Pull request merged**